docs/source/data-access-api/openapi.yml

```
@@ -0,0 +1,159 @@
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+
+  '/download/{filepath-or-filename}':
+    get:
+      tags:
+        - Download
+      summary: Download a file from the SDC
+      operationId: download
+      parameters:
+        - in: path
+          name: filepath-or-filename
+          description: |
+            The filename to download (e.g. ``imap_codice_l1a_lo_20260101_20260102_v01-01.cdf``, optionally including the
+            full filepath (e.g. ``codice/l1a/2026/01/imap_codice_l1a_lo_20260101_20260102_v01-01.cdf``). The filename
+            must be a valid IMAP filename that follows the conventions described
+            `here <https://imap-processing.readthedocs.io/en/latest/development-guide/style-guide/naming-conventions.html#data-product-file-naming-conventions>`_.
+          required: true
+          schema:
+            type: string
+      responses:
+        '302':
+          description: Successful download
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+                  format: uri
+        '400':
+          description: Unsuccessful download
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+                  format: uri
+
+  '/query':
+    get:
+      tags:
+        - Query
+      summary: Query for file metadata
+      operationId: query
+      parameters:
+        - in: query
+          name: instrument
+          description: |
+            The instrument of interest (e.g. ``mag``). Supported instruments are listed
+            `here <https://imap-processing.readthedocs.io/en/latest/development-guide/style-guide/naming-conventions.html#data-product-file-naming-conventions>`_.
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: data_level
+          description: |
+            The level of data product (e.g. ``l1a``). Supported levels are listed
+            `here <https://imap-processing.readthedocs.io/en/latest/development-guide/style-guide/naming-conventions.html#data-product-file-naming-conventions>`_.
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: descriptor
+          description: |
+            The descriptor of interest (e.g. ``burst``). Supported descriptors are listed
+            `here <https://imap-processing.readthedocs.io/en/latest/development-guide/style-guide/naming-conventions.html#data-product-file-naming-conventions>`_.
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: start_date
+          description: The start date of observation in the format ``YYYYMMDD``.
```

```suggestion
          description: Search for all files with a start date on or after this time, in the format ``YYYYMMDD``.
```
and for the end_date:
`description: Search for all files with a start date on or before this time, in the format ``YYYYMMDD``.`

Do we want to follow the date hint?
https://swagger.io/docs/specification/data-models/data-types/
> date – full-date notation as defined by [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6), for example, 2017-07-21

Looks like this date isn't the suggested with a "-" if we care about that.

---

Good question. I think I will punt on this for now, but bring it up for discussion when we go over the API.